pkg/controllerhelpers/apps.go

```diff
@@ -25,6 +25,14 @@ func IsStatefulSetRolledOut(sts *appsv1.StatefulSet) (bool, error) {
 		return false, nil
 	}
 
+	if sts.Status.AvailableReplicas < *sts.Spec.Replicas {
+		return false, nil
+	}
+
+	if sts.Status.Replicas < *sts.Spec.Replicas {
```

Comment:
how can there be less replicas then availableReplicas checked in the previous steps?

Reply:
it can't, rearranged and changed to require equality.